### PR TITLE
Add packet info (type and id) attribute to `aiex.npu.dma_mempcy_nd`

### PR DESCRIPTION
**Pull request by erwei-xilinx**

### CODE REVIEW

**Comment by fifield** [comment]
The new attribute could be useful elsewhere:
```mlir
// replace this,
aie.dma_bd_packet(0xA, 0xB)
aie.dma_bd($buf0 : memref<..>)

// with this?
aie.dma_bd($buffer : memref<..>) { packet = <pkt_id = 0xA, pkt_type = 0xB>)
```

File: lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp [hunk 1]
```diff
@@ -370,13 +370,14 @@ struct DmaToNpuPattern : OpConversionPattern<NpuDmaMemcpyNdOp> {
     buffer_offset = IntegerAttr::get(i32ty, offset);
 
     // enable_packet
+    if (auto packetInfo = op.getPacket()) {
+      enable_packet = IntegerAttr::get(i32ty, 1);
+      packet_type = IntegerAttr::get(i32ty, packetInfo->getPktType());
+      packet_id = IntegerAttr::get(i32ty, packetInfo->getPktId());
+    }
```
Comment (fifield) [inline lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp:377]: This needs a test

Comment (erwei-xilinx) [reply lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp:377]: Sure. This PR contains new mlir tests to check the new format and verifier.

Re board test, I will land this PR, and then convert its SHIM->Memtile flow to packet flow, as board test for this feature. https://github.com/Xilinx/mlir-aie/pull/1669

**Comment by erwei-xilinx** [comment]
> The new attribute could be useful elsewhere:
> 
> ```mlir
> // replace this,
> aie.dma_bd_packet(0xA, 0xB)
> aie.dma_bd($buffer : memref<..>)
> 
> // with this?
> aie.dma_bd($buffer : memref<..>) { packet = <pkt_id = 0xA, pkt_type = 0xB>)
> ```

Makes great sense to me. Can come in a separate PR.

File: lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp (hunk 1 above)
Comment (fifield) [reply lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp:377]: > Sure. This PR contains new mlir tests to check the new format and verifier.
> 

It also needs a mlir test for the `aie-dma-to-npu` pass update in e.g. `test/Conversion/DmaToNpu/dma_to_npu.mlir`

Comment (erwei-xilinx) [reply lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp:377]: Ah got it. New test added in commit [Add test to show aiex.npu.dma lowering with packet](https://github.com/Xilinx/mlir-aie/pull/1671/commits/61d20fb8bc135d3782eb210f3dd9b3eeba9ccbba)

Comment (erwei-xilinx) [reply lib/Dialect/AIEX/Transforms/AIEDmaToNpu.cpp:377]: Board test added in commit [Packet header from shim dma board test](https://github.com/Xilinx/mlir-aie/pull/1671/commits/15eb5058d31b689e1f3888413732646be2a876ec)

**Review by fifield — APPROVED** [review]

**Comment by erwei-xilinx** [comment]
> > The new attribute could be useful elsewhere:
> > ```mlir
> > // replace this,
> > aie.dma_bd_packet(0xA, 0xB)
> > aie.dma_bd($buffer : memref<..>)
> > 
> > // with this?
> > aie.dma_bd($buffer : memref<..>) { packet = <pkt_id = 0xA, pkt_type = 0xB>)
> > ```
> 
> Makes great sense to me. Can come in a separate PR.

Added in this PR: https://github.com/Xilinx/mlir-aie/pull/1674

**Pull request merged**